safekeeper/src/receive_wal.rs

```diff
@@ -246,9 +351,11 @@ impl WalAcceptor {
     /// it must mean that network thread terminated.
     async fn run(&mut self) -> anyhow::Result<()> {
         // Register the connection and defer unregister.
-        self.tli.on_compute_connect().await?;
-        let _guard = ComputeConnectionGuard {
+        let walreceiver_guard = self.tli.get_walreceivers().register();
+        self.tli.update_status_notify().await?;
```

Comment:
Can we move these 2 lines below the `compute_conn_guard`? Then we will not need to add `walreceiver_guard` to the `ComputeConnectionGuard` and use `ManuallyDrop`, but it will work the same.

Reply:
Yes, variables are dropped in reverse declaration order, that would be easier.